src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -603,6 +607,11 @@ MulticopterPositionControl::Run()
 				_control.resetIntegral();
 				// reactivate the task which will reset the setpoint to current state
 				_flight_tasks.reActivate();
+				_hover_thrust_estimator.reset();
+
+			} else if (_takeoff.getTakeoffState() >= TakeoffState::flight) {
```

Comment:
Do we need to handle landing as well to make sure it doesn't get corrupted? Or will this work for the landing case too?

Reply:
Good point, i would need to check that

Reply:
Why is this in the else case? Would you want to run it if `PX4_ISFINITE(setpoint.thrust[2])`?

Reply:
@bresch did you check this?

Reply:
@mhkabir Yes, due to the high inconsistency at touchdown, the measurements are rejected until the recovery kicks-in and we should be able to detect landing in the mean time (using that inconsistency as a ground contact check).
About the "Why is this in the else case? Would you want to run it if PX4_ISFINITE(setpoint.thrust[2])?", I placed it in that else because it gets reset when on ground in the `if` above, so we don't need to do a double check. It might not be ideal but I can still change that when I use the output of the filter for feedforward and land detection.